### PR TITLE
E2E test - check pod.Status.Phase when greping through pod logs

### LINKED ISSUE #14462: "Spark should start spark master, driver and workers" e2e test continuously failing
This test hasn't passed a single time in the last week on GCE.

The failure typically looks like the following:
```
15/09/24 00:51:13 INFO Utils: Successfully started service 'WorkerUI' on port 8081.
15/09/24 00:51:13 INFO WorkerWebUI: Started WorkerWebUI at http://10.245.1.6:8081
15/09/24 00:51:13 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/09/24 00:51:13 INFO Worker: Successfully registered with master spark://spark-master:7077

Sep 23 17:51:18.244: INFO: Waiting up to 5m0s for pod spark-worker-controller-zcr56                           status to be running
Sep 23 17:51:18.248: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:51:23.252: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:51:28.255: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:51:33.259: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:51:38.263: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:51:43.267: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:51:48.270: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:51:53.274: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:51:58.277: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:03.280: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:08.284: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:13.288: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:18.291: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:23.295: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:28.298: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:33.302: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:38.305: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:43.308: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:48.311: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:53.315: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:52:58.319: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:03.322: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:08.326: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:13.330: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:18.333: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:23.336: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:28.339: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:33.342: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:38.346: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:43.349: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:48.352: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:53.355: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:53:58.359: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:03.362: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:08.365: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:13.369: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:18.372: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:23.376: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:28.379: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:33.382: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:38.385: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:43.389: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:48.393: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:53.396: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:54:58.400: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:03.406: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:08.411: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:13.414: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:18.418: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:23.421: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:28.426: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:33.429: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:38.432: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:43.436: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:48.440: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:53.443: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:55:58.446: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:56:03.450: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:56:08.453: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
Sep 23 17:56:13.457: INFO: Get pod spark-worker-controller-zcr56                           in namespace 'e2e-tests-examples-s7ngd' failed, ignoring for 5s. Error: pods "spark-worker-controller-zcr56" not found
[AfterEach] Examples e2e
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/examples.go:69
STEP: Destroying namespace for this suite e2e-tests-examples-s7ngd

• Failure [428.117 seconds]
Examples e2e
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/examples.go:528
  [Skipped][Example]Spark
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/examples.go:203
    should start spark master, driver and workers [It]
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/examples.go:202

    Expected error:
        <*errors.errorString | 0xc208359880>: {
            s: "gave up waiting for pod 'spark-worker-controller-zcr56' to be 'running' after 5m0s",
        }
        gave up waiting for pod 'spark-worker-controller-zcr56' to be 'running' after 5m0s
    not to have occurred

    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/examples.go:579
```

@mwielgus @gurvindersingh not sure which of you want to take this

cc @kubernetes/goog-testing for good measure too

### PR DESCRIPTION
**Pull request by mwielgus**
Fixes #14462

### CODE REVIEW

**Comment by mwielgus** [comment]
cc: @ixdy

File: test/e2e/examples.go [hunk 1]
```diff
@@ -562,22 +562,26 @@ func prepareResourceWithReplacedString(inputFile, old, new string) string {
...
-	var pods *api.PodList
-	var err error
+	pods := []*api.Pod{}
 	for t := time.Now(); time.Since(t) < podListTimeout; time.Sleep(poll) {
-		pods, err = c.Pods(ns).List(labels.SelectorFromSet(labels.Set(map[string]string{selectorKey: selectorValue})), fields.Everything())
+		podList, err := c.Pods(ns).List(labels.SelectorFromSet(labels.Set(map[string]string{selectorKey: selectorValue})), fields.Everything())
 		Expect(err).NotTo(HaveOccurred())
-		if len(pods.Items) > 0 {
+		for _, pod := range podList.Items {
+			if pod.Status.Phase == api.PodPending || pod.Status.Phase == api.PodRunning {
```
Comment (pmorie) [inline test/e2e/examples.go]: What about Succeeded?

Comment (mwielgus) [inline test/e2e/examples.go]: All test cases in examples.go focus on running pods. Success pod may (I'm not sure) disappear just like Terminated pods causing problems to log greping.

**Comment by mwielgus** [comment]
@k8s-bot test this please

**Comment by piosz** [comment]
LGTM

**Comment by piosz** [comment]
Shippable is flaky

**Pull request merged**